shared/profile/index.js

```diff
@@ -4,25 +4,32 @@ import {connect} from 'react-redux'
 import Render from './render'
 import type {Props} from './render'
 import flags from '../util/feature-flags'
-import {getProfile} from '../actions/tracker'
-
-// TEMP
-const usernames = ['chromakode', 'max', 'jzila', 'mikem', 'strib', 'zanderz', 'gabrielh', 'chris',
-  'songgao', 'patrick', 'awendland', 'marcopolo', 'akalin', 'cjb', 'oconnor663', 'cbostrander',
-  'alness', 'chrisnojima', 'jinyang', 'cecileb']
-const username = usernames[Math.floor(Math.random() * usernames.length)]
-// TEMP
+import {getProfile, updateTrackers} from '../actions/tracker'
+import {routeAppend, navigateUp} from '../actions/router'
 
 class Profile extends Component<void, Props, void> {
-  static parseRoute () {
+  static parseRoute (currentPath, uri) {
     return {
-      componentAtTop: {title: 'Profile'},
+      componentAtTop: {
+        title: 'Profile',
+        props: {
+          username: currentPath.get('username'),
+          profileRoot: uri.count() && uri.last().get('path') === 'root',
+        },
+      },
       subRoutes: {},
     }
   }
 
   componentDidMount () {
-    this.props.refresh()
+    this.props.refresh(this.props.username)
+  }
+
+  shouldComponentUpdate (nextProps, nextState) {
```

Comment:
with the nav we have on desktop we don't actually keep the true stack of components , so when you push a profile onto another profile it actually just replaces the data. this makes this all slightly more tricky. we have to keep certain props in the props vs just figuring this out in connect